pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -462,6 +467,74 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileBuildRun) updateBuildRunUsingTaskRunCondition(ctx context.Context, buildRun *buildv1alpha1.BuildRun, taskRun *v1beta1.TaskRun, trCondition *apis.Condition) error {
+	var reason, message string = trCondition.Reason, trCondition.Message
+
+	switch v1beta1.TaskRunReason(reason) {
+	case v1beta1.TaskRunReasonTimedOut:
+		reason = "BuildRunTimeout"
+		message = fmt.Sprintf("BuildRun %s failed to finish within %s",
+			taskRun.GetLabels()[buildv1alpha1.LabelBuildRun],
+			taskRun.Spec.Timeout.Duration,
+		)
+
+	case v1beta1.TaskRunReasonFailed:
+		if taskRun.Status.CompletionTime != nil {
+			var pod corev1.Pod
+			if err := r.client.Get(ctx, types.NamespacedName{Namespace: taskRun.Namespace, Name: taskRun.Status.PodName}, &pod); err != nil {
+				return err
+			}
+
+			// Since the container status list is not sorted, as a quick workaround mark all failed containers
+			var failures = make(map[string]struct{})
+			for _, containerStatus := range pod.Status.ContainerStatuses {
+				if containerStatus.State.Terminated != nil && containerStatus.State.Terminated.ExitCode != 0 {
+					failures[containerStatus.Name] = struct{}{}
+				}
+			}
+
+			// Find the first container that failed
+			var failedContainer *corev1.Container
+			for i, container := range pod.Spec.Containers {
+				if _, has := failures[container.Name]; has {
+					failedContainer = &pod.Spec.Containers[i]
+					break
+				}
+			}
+
+			if failedContainer != nil {
+				message = fmt.Sprintf("buildrun step failed in pod %s, for detailed information: kubectl --namespace %s logs %s --container=%s",
```

Comment:
Perhaps we bubble up terminated state message from the pod here, rather than a "look at the logs" message? https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#containerstateterminated-v1-core

Reply:
@adambkaplan interesting, let me play with that tomorrow, and then I can give you a proper answer. Note: This constructed message is very similar to what Tekton does on a failed Taskrun, it provides you some metadata + the `kubectl` cmd to execute.

Reply:
@adambkaplan I took a look on this, this will not work, while Tekton does not exposes internal results. They override the message with `InternalTektonResult`, this will look like:
```
    state:
      terminated:
        containerID: containerd://fd0627262107d2339abc75261afd4ca66a91f1b6a3f19e2a3ae68a9ff817138e
        exitCode: 1
        finishedAt: "2020-12-14T17:06:19Z"
        message: '[{"key":"StartedAt","value":"2020-12-14T17:06:19.706Z","type":"InternalTektonResult"}]'
        reason: Error
        startedAt: "2020-12-14T17:06:13Z"
```

I think the current approach is good for us, also it will help users to move into the right direction, which is "getting the logs of the failed step".